helixbass/tree-sitter-lint
# LSP trace files

In this PR:
- set up logging trace files from LSP server for non-fixing runs
- cache instantiated rules/aggregated queries in LSP server to speed up non-initial LSP non-fixing runs
- wire up JS linting in the LSP server 

Not in this PR:
- doing the same thing for LSP fixing runs presumably 

To test:
I set up a config file for the ESLint repo and turned on the ESLint-builtin "all" config and recorded some trace files (specifically by setting `TRACE_CHROME=1` in my editor's LSP server config) and was able to see that (as expected) query parsing was the "slow part" and that after caching the parsed aggregated queries the subsequent lint runs were significantly faster
